api/routes/auth/login.endpoint.js

```diff
@@ -1,53 +1,53 @@
 const jwt = require('jsonwebtoken');
-const { request, getFullPath } = require('../../utils.endpoint');
+const qs = require('querystring');
+const { api, buildForm } = require('../../endpoint-tests/utils');
 
 describe('login nonce endpoint | /auth/login/nonce', () => {
-  const url = getFullPath('/auth/login/nonce');
+  const url = '/auth/login/nonce';
 
   it('with no username', async () => {
-    const {
-      response: { statusCode }
-    } = await request.post(url);
-    expect(statusCode).toEqual(400);
+    const response = await api.post(url).then(res => res);
+    expect(response.status).toEqual(400);
   });
 
   it('with a username', async () => {
-    const {
-      response: { statusCode },
-      body
-    } = await request.post(url, { json: { username: 'test user' } });
-    expect(statusCode).toEqual(200);
+    const response = await api
+      .post(url, { username: 'test user' })
+      .then(res => res);
+    expect(response.status).toEqual(200);
 
-    const token = jwt.decode(body.nonce);
+    const token = jwt.decode(response.data.nonce);
 
-    expect(body.nonce).toMatch(/[^.]+\.[^.]+\.[^.]+/i);
+    expect(response.data.nonce).toMatch(/[^.]+\.[^.]+\.[^.]+/i);
     expect(token).toMatchObject({ username: 'test user' });
     expect(token.exp).toEqual(token.iat + 3);
   });
 });
 
 describe('login endpoint | /auth/login', () => {
-  const nonceUrl = getFullPath('/auth/login/nonce');
+  const nonceUrl = '/auth/login/nonce';
   const nonceForUsername = async username => {
-    const {
-      body: { nonce }
-    } = await request.post(nonceUrl, { json: { username } });
-    return nonce;
+    const response = await api.post(nonceUrl, { username }).then(res => res);
+    return response.data.nonce;
   };
 
-  const url = getFullPath('/auth/login');
+  const url = '/auth/login';
 
   it('with no post body at all', async () => {
-    const { response } = await request.post(url);
-    expect(response.statusCode).toEqual(400);
+    const response = await api.post(url).then(res => res);
+    expect(response.status).toEqual(400);
   });
 
-  it('proper response headers', async () => {
-    const { response } = await request.post(url);
-    const { headers } = response;
-
-    expect(headers.vary).toEqual('Origin');
-    expect(headers['access-control-allow-credentials']).toEqual('true');
+  xit('proper response headers', async () => {
+    // I'm unable to pick up these headers w/ axios
+    const response = await api.post(url).then(res => res);
+    // console.log(response.headers);  // -> { 'cache-control': 'private, no-cache' }
+    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Vary
+    expect(response.headers.vary).toEqual('Origin');
+    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials
+    expect(response.headers['access-control-allow-credentials']).toEqual(
+      'true'
+    );
```

Comment:
Hey @mgwalker, I switched our endpoint tests from [`node request`](https://github.com/request/request) to axios since the request library was deprecated in February. I'm unable to see these headers. 

Are they important? I can't find in the codebase where they should be set by the server. Halp! :)

Reply:
They were, but I don't think they are anymore. I'll have to dig into it again but I think this header was necessary to allow Javascript to send authenticated requests back to the API later with cookies. Since the requests are authenticated with a token in a header ***and*** that token is sent back in a response body (instead of being sent in a cookie or header), I think we don't need this header anymore.

However, we also need to remove the `withCredentials: true` configuration from Axios on the frontend. If the `withCredentials` property of the request is true, then that header must also be set; otherwise, the browser won't deliver the results into the JS code.